projects/media-viewer/src/lib/toolbar/side-bar/outline-view/outline-view.component.ts

```diff
@@ -0,0 +1,40 @@
+import * as pdfjsLib from 'pdfjs-dist';
+import { AfterContentInit, Component, ElementRef, Input, OnChanges, SimpleChanges, ViewChild } from '@angular/core';
+
+@Component({
+  selector: 'mv-outline-view',
+  templateUrl: './outline-view.component.html',
+  styleUrls: ['../../../styles/main.scss'],
+})
+export class OutlineViewComponent implements OnChanges, AfterContentInit {
+
+  @Input() url: string;
+  @ViewChild('outlineContainer') outlineContainer: ElementRef;
+
+  outline = [];
+
+  constructor() {
+  }
+
+  async ngAfterContentInit() {
+    this.renderOutlineView();
+  }
+
+  async ngOnChanges(changes: SimpleChanges) {
+    if (changes.url) {
+      this.outline = [];
+      await this.renderOutlineView();
+    }
+  }
+
+  async renderOutlineView() {
+    const pdf = await pdfjsLib.getDocument({
```

Comment:
Are you sure we should be fetching the entire document again? Can't we store it someone once it gets loaded?

Reply:
the issue is the sidebar and main viewer are loaded at the same time, so this becomes a problem with getting the document.